xblock/fields.py

```diff
@@ -646,6 +650,24 @@ def is_set_on(self, xblock):
 
     def __hash__(self):
         return hash(self.name)
+    
+    @classmethod
+    def Query(cls, field_name):
+        """
+        Return a instance of Query which allow ad-hoc queries to access shared fields. 
+        """
+        return Query(field_name)
```

Comment:
Doesn't `Query` need to get passed `cls`, so that it knows what type of field it's being declared as? (Although, thinking about it now, when does it resolve issues where the type of `cls` and the type of the field identified by `field_name` differ?)

Also, `field_name` isn't enough to uniquely identify a block, is it? How do you reference a field in another XBlock type? Does the `Query` definition need to have the block type and the field name passed as arguments?